apps/src/componentLibrary/modal/Modal.tsx

```diff
@@ -1,86 +1,145 @@
+  /** Modal color mode */
+  mode?: 'light' | 'dark';
   /** Custom class name */
   className?: string;
   /** Modal onClose handler */
   onClose?: () => void;
   /** Modal close button aria label */
   closeLabel?: string;
-  /** Modal color */
-  color?: 'light' | 'dark';
+  /** Modal image url */
+  imageUrl?: string;
+  /** Modal image alt */
+  imageAlt?: string;
+  /** Modal image placement */
+  imagePlacement?: 'top' | 'inline';
 }
 
 /**
  * ## Production-ready Checklist:
- *  * (?) implementation of component approved by design team;
- *  * (?) has storybook, covered with stories and documentation;
- *  * (?) has tests: test every prop, every state and every interaction that's js related;
+ *  * (✔) implementation of component approved by design team;
+ *  * (✔) has storybook, covered with stories and documentation;
+ *  * (✔) has tests: test every prop, every state and every interaction that's js related;
  *  * (see apps/test/unit/componentLibrary/ModalTest.tsx)
  *  * (?) passes accessibility checks;
  *
- * ###  Status: ```WIP```
+ * ###  Status: ```Ready for dev```
  *
  * Design System: Modal Component.
- * Renders Alert to notify user about something.
+ * Renders Modal window that user should interact with.
  */
 const Modal: React.FunctionComponent<ModalProps> = ({
   title,
-  color = 'light',
-  content,
-  showSecondaryButton,
+  description,
+  primaryButtonProps,
+  secondaryButtonProps,
+  mode = 'light',
   className,
+  customContent,
+  customBottomContent,
   onClose,
-  closeLabel = 'Close dialog',
+  closeLabel = 'Close modal',
+  imageUrl,
+  imageAlt,
+  imagePlacement = 'top',
   ...HTMLAttributes
 }) => {
+  const modalRef = useRef<HTMLDivElement>(null);
+
+  useBodyScrollLock(true);
+  useFocusTrap(modalRef);
+  useEscapeKeyHandler(onClose);
+
   return (
-    <div
-      className={classnames(
-        moduleStyles.dialog,
-        moduleStyles[`dialog-${color}`],
-        className
-      )}
-      role="dialog"
-      {...HTMLAttributes}
-    >
-      <div>
-        <span>{title}</span>
-        <span className={moduleStyles.dialogContent}>{content}</span>
-        {showSecondaryButton && (
+    <div role="presentation" className={moduleStyles.modalOverlay}>
+      <div
+        role="dialog"
+        ref={modalRef}
+        aria-modal
+        aria-label={title}
+        aria-describedby="dsco-dialog-description"
+        className={classnames(
+          moduleStyles.modal,
+          moduleStyles[`modal-${mode}`],
+          className
+        )}
+        {...HTMLAttributes}
+      >
+        <div className={moduleStyles.modalTitleSection}>
+          <Heading3>{title}</Heading3>
+        </div>
+        <hr />
+        <div
+          className={classnames(
+            moduleStyles.modalContentSection,
+            moduleStyles[`modalContentSection-${imagePlacement}-imagePlacement`]
+          )}
+        >
+          {imageUrl && <img src={imageUrl} alt={imageAlt || 'modal-image'} />}
```

Comment:
It's best to have empty alt text if the image is descriptive, which I assume this will be most of the time. Can the default be swapped with empty alt text instead? cc @hannahbergam as our accessibility expert
```suggestion
          {imageUrl && <img src={imageUrl} alt={imageAlt || ""} />}
```

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img

> The alt attribute holds a textual replacement for the image, which is mandatory and incredibly useful for accessibility — screen readers read the attribute value out to their users so they know what the image means. Alt text is also displayed on the page if the image can't be loaded for some reason: for example, network errors, content blocking, or link rot.

There're no plans on adding figcaption to modal right now. No matter what the content of the image will be, if user won't be able to see it - user won't be able to understand it's content since screen reader won't have where to take image description from.

Thus, imgAlt is required here

Reply:
if the image is decorative rather than informative, having a default alt text is more disruptive to users of screen readers than leaving it empty. What does "modal-image" convey to the user, and what would the experience be like to hear that alongside the informative text in the modal?

https://www.w3.org/WAI/tutorials/images/

Reply:
+1 to what Ryan said above. Here's more from that resource:

> Decorative images don’t add information to the content of a page. For example, the information provided by the image might already be given using adjacent text, or the image might be included to make the website more visually attractive.
> 
> In these cases, a null (empty) alt text should be provided (alt="") so that they can be ignored by assistive technologies, such as screen readers. Text values for these types of images would add audible clutter to screen reader output or could distract users if the topic is different from that in adjacent text. Leaving out the alt attribute is also not an option because when it is not provided, some screen readers will announce the file name of the image instead.

Reply:
> What does "modal-image" convey to the user,
I agree it's not the best alt, added it here just to make sure that there'll always be an alt for the image.

Reply:
Thanks for the context! Based on your suggestions and w3 article I agree to @kelbyhawn's proposal, adding `alt={imageAlt || ""}`

Reply:
<img width="1078" alt="Знімок екрана 2024-11-26 о 21 22 02" src="https://github.com/user-attachments/assets/5164480b-cb8b-4f8a-8ad5-2c1a81882509">
Here how it will be interpreted in html code